simulator/simple_planning_simulator/include/simple_planning_simulator/utils/mechanical_controller.hpp

```diff
@@ -0,0 +1,204 @@
+
+using DelayBuffer = std::deque<std::pair<double, double>>;
+using DelayOutput = std::pair<std::optional<double>, DelayBuffer>;
+
+DelayOutput delay(
+  const double signal, const double delay_time, const DelayBuffer & buffer,
+  const double elapsed_time);
+
+double sign(const double x);
+
+double apply_limits(
+  const double current_angle, const double previous_angle, const double angle_limit,
+  const double rate_limit, const double dt);
+
+double feedforward(const double input_angle, const double ff_gain);
+
+double polynomial_transform(
+  const double torque, const double speed, const double a, const double b, const double c,
+  const double d, const double e, const double f, const double g, const double h);
+
+struct PIDControllerParams
+{
+  double kp{0.0};
+  double ki{0.0};
+  double kd{0.0};
+};
+
+struct PIDControllerState
+{
+  double integral{0.0};
+  double error{0.0};
+};
+
+class PIDController
+{
+public:
+  PIDController(const double kp, const double ki, const double kd);
+
+  [[nodiscard]] double compute(
+    const double error, const double integral, const double prev_error, const double dt) const;
+
+  void update_state(const double error, const double dt);
+
+  [[nodiscard]] PIDControllerState get_state() const;
+
+  void clear_state();
+
+private:
+  double kp_{0.0}, ki_{0.0}, kd_{0.0};
+  PIDControllerState state_;
+};
+
+struct SteeringDynamicsParams
+{
+  double angular_position{0.0};
+  double angular_velocity{0.0};
+  double inertia{0.0};
+  double damping{0.0};
+  double stiffness{0.0};
+  double friction{0.0};
+  double dead_zone_threshold{0.0};
+};
+
+struct SteeringDynamicsState
+{
+  double angular_position{0.0};
+  double angular_velocity{0.0};
+  bool is_in_dead_zone{false};
+};
+
+struct SteeringDynamicsDeltaState
+{
+  double d_angular_position{0.0};
+  double d_angular_velocity{0.0};
+};
+
+/**
+ * @brief SteeringDynamics class
+ * @details This class calculates the dynamics which receives the steering torque and outputs the
+ * steering tire angle. The steering system is modeled as a spring-damper system with friction and
+ * dead zone.
+ */
+class SteeringDynamics
+{
+public:
+  SteeringDynamics(
+    const double angular_position, const double angular_velocity, const double inertia,
+    const double damping, const double stiffness, const double friction,
+    const double dead_zone_threshold);
+
+  [[nodiscard]] bool is_in_dead_zone(
+    const SteeringDynamicsState & state, const double input_torque) const;
+
+  [[nodiscard]] SteeringDynamicsDeltaState calc_model(
+    const SteeringDynamicsState & state, const double input_torque) const;
+
+  void set_state(const SteeringDynamicsState & state);
+
+  [[nodiscard]] SteeringDynamicsState get_state() const;
+
+  void clear_state();
+
+  void set_steer(const double steer);
+
+private:
+  SteeringDynamicsState state_;
+  double inertia_{0.0};
+  double damping_{0.0};
+  double stiffness_{0.0};
+  double friction_{0.0};
+  double dead_zone_threshold_{0.0};
```

Comment:
```suggestion
  const double inertia_;
  const double damping_;
  const double stiffness_;
  const double friction_;
  const double dead_zone_threshold_;
```

Reply:
fixed in [update from suggestions](https://github.com/autowarefoundation/autoware.universe/pull/9300/commits/ed8093d903604c640d5739a715358ab5c2e2bcce)